### PR TITLE
Add vlan remote mechanism support

### PR DESCRIPTION
**Pull request by ljkiraly**
This PR contains the remote vlan mechanism support enhancement for vpp.
Related PR: api#112

### CODE REVIEW

**Comment by edwarnicke** [comment]
@ljkiraly I think this needs rebasing now that the api changes are merged?

File: pkg/networkservice/mechanisms/vlan/client.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	vppConn api.Connection
+}
+
+// NewClient returns a VLAN client chain element
+func NewClient(vppConn api.Connection, resourceLock sync.Locker, resourcePool pciaddresspool.PCIAddressPool) networkservice.NetworkServiceClient {
+	return chain.NewNetworkServiceClient(
+		l2vtr.NewClient(vppConn),
+		&vlanClient{vppConn: vppConn},
+		linkinit.NewClient(vppConn),
+		pciaddresspool.NewClient(resourceLock, resourcePool),
```
Comment (edwarnicke) [inline pkg/networkservice/mechanisms/vlan/client.go:58]: I'm confused as to why pciaddresspool is here, and more so, why it's pulling in sdk-sriov into this PR...

Comment (ljkiraly) [reply pkg/networkservice/mechanisms/vlan/client.go:58]: It's a simplified PCI address pool which maps the service domain to an interface name using the PCI addresses.

Comment (edwarnicke) [reply pkg/networkservice/mechanisms/vlan/client.go:58]: Ah... OK... now I am remembering... and it pulls through sdk-sriov ?  Is there anyway to have it not do so simply?

Comment (edwarnicke) [reply pkg/networkservice/mechanisms/vlan/client.go:58]: @ljkiraly If I am remembering correctly, the root issue here is "How to know which interface to bind to"  For tunnels we do this by passing a TunnelIP in as a parameter, and discovering which one to pass in in the cmd-forwarder-vpp.  What parameter might we pass in at the top here?

Comment (ljkiraly) [reply pkg/networkservice/mechanisms/vlan/client.go:58]: No, the resourcepool in sdk-sriov is more complex and also a PCI pool is needed to get it working. To initialize the pools require additional parameters which are not used here. So the sdk-sriov is only used to process the configuration file. The pool and the mapping is in sdk-vpp.

Comment (ljkiraly) [reply pkg/networkservice/mechanisms/vlan/client.go]: I moved the domain to device config to cmd-forwarder-vpp and pass the map to clients in sdk-vpp

**Comment by ljkiraly** [comment]
Still I have the following topics to resolve:
- The service domain should be get from Connection.NetworkServiceName. Currently the service name does not contains the domain. (e.g finance-bridge@domain.1 is broken down to service name: finance-bridge and service domain: domain.1)

- Each interface in NSC has different MAC address. Some infrastructure the MAC filtering drops packets to unknown MAC addresses. The 'local mechanism' interface added to NSC namespace needs to have the same MAC address as the NIC that the vlan is going out.

- Should work if the base interface is already added to VPP

File: pkg/networkservice/mechanisms/vlan/client.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	"github.com/networkservicemesh/sdk/pkg/tools/postpone"
+
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/hwaddress"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/l2vtr"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/linkinit"
+	"github.com/networkservicemesh/sdk-vpp/pkg/tools/ifindex"
+)
+
+const (
+	serviceDomainLabel = "serviceDomain"
```
Comment (edwarnicke) [inline pkg/networkservice/mechanisms/vlan/client.go:47]: Why are we doing this with a label rather than as part of the networkservice name?

Comment (ljkiraly) [reply pkg/networkservice/mechanisms/vlan/client.go:42]: If the domain name is added to the service name the client must be configured to connect to this service by specifying the domain name also. The client configuration should not contain information about domain or vlan ID.

Comment (edwarnicke) [reply pkg/networkservice/mechanisms/vlan/client.go:42]: @ljkiraly Network Service Mesh already has network service domain as part of the Network Service Name.

So for example if the client wants 'foo@radio.example.com' the annotation would be:

```
networkservicemesh.io: kernel://foo@radio.example.com
```

The user doesn't have to know anything about the vlan ID

Comment (ljkiraly) [reply pkg/networkservice/mechanisms/vlan/client.go:42]: The NSE is running locally, and no other domain involved. So the service domain in my case is just a label. It is used to identify the physical device to use. What about renaming it to domain-label?

**Pull request merged**